validator/specs/lib/validateCmptSpec.js

```diff
@@ -0,0 +1,279 @@
+
+    it('returns false if the cmpt has an invalid pnts inner tile', function() {
+        expect(validateCmpt(createCmptInvalidPnts()).result).toBe(false);
+    });
+
+    it('returns false if the cmpt contains any invalid inner tiles', function() {
+        expect(validateCmpt(createCmptInvalidCombination()).result).toBe(false);
+    });
+});
+
+var b3dmHeaderSize = 24;
+var i3dmHeaderSize = 32;
+var pntsHeaderSize = 28;
+var cmptHeaderSize = 16;
+var unknownHeaderSize = 12;
+var sizeU32Int = 4;
+
+function createB3dmTile(version, byteLength) {
+    var byteOffset = 0;
+
+    var b3dmTile = new Buffer(b3dmHeaderSize);
+    b3dmTile.write('b3dm', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    b3dmTile.writeUInt32LE(version, byteOffset); // version
+    byteOffset += sizeU32Int;
+    b3dmTile.writeUInt32LE(byteLength, byteOffset); // byteLength
+
+    return b3dmTile;
+}
+
+function createI3dmTile(version, byteLength, gltfFormat) {
+    var byteOffset = 0;
+
+    var i3dmTile = new Buffer(i3dmHeaderSize);
+    i3dmTile.write('i3dm', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    i3dmTile.writeUInt32LE(version, byteOffset); // version
+    byteOffset += sizeU32Int;
+    i3dmTile.writeUInt32LE(byteLength, byteOffset); // byteLength
+    byteOffset += sizeU32Int;
+    byteOffset += sizeU32Int; // skip featureTableJSONByteLength
+    byteOffset += sizeU32Int; // skip featureTableBinaryByteLength
+    byteOffset += sizeU32Int; // skip batchTableJSONByteLength
+    byteOffset += sizeU32Int; // skip batchTableBinaryByteLength
+    i3dmTile.writeUInt32LE(gltfFormat, byteOffset); // gltfFormat: 0 - url
+
+    return i3dmTile;
+}
+
+function createCmptTile(version, byteLength, tilesLength) {
+    var byteOffset = 0;
+
+    var cmptTile = new Buffer(cmptHeaderSize);
+    cmptTile.write('cmpt', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(version, byteOffset); // version
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(byteLength, byteOffset); // byteLength
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(tilesLength, byteOffset); // tilesLength
+
+    return cmptTile;
+}
+
+function createPntsTile(version, byteLength) {
+    var byteOffset = 0;
+
+    var pntsTile = new Buffer(pntsHeaderSize);
+    pntsTile.write('pnts', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    pntsTile.writeUInt32LE(version, byteOffset); // version
+    byteOffset += sizeU32Int;
+    pntsTile.writeUInt32LE(byteLength, byteOffset); // byteLength
+
+    return pntsTile;
+}
+
+function createUnknownTile(version, byteLength) {
+    var byteOffset = 0;
+
+    var unknownTile = new Buffer(12);
+    unknownTile.write('xxxx', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    unknownTile.writeUInt32LE(version, byteOffset); // version
+    byteOffset += sizeU32Int;
+    unknownTile.writeUInt32LE(byteLength, byteOffset); // byteLength
+
+    return unknownTile;
+}
+
+function createShortHeader() {
```

Comment:
This can also be replaced by calling `createCmpt` with an empty array of tile buffers, and then `buffer.slice(cmptHeaderSize - 4)`